File: Variable firmware/backend/api/delay.py
import json
from typing import Dict, Any
from http import HTTPStatus

# In-memory state (⚠️ resets on cold start – use Redis/DB in real app)
_state = {"delay": 1000}  # Default: 1000 ms

MIN_DELAY = 500
MAX_DELAY = 5000

def clamp_delay(value: int) -> int:
    return max(MIN_DELAY, min(MAX_DELAY, value))

def handler(event: Dict[str, Any], context) -> Dict[str, Any]:
    method = event.get("httpMethod", "")
    
    if method == "GET":
        return {
            "statusCode": HTTPStatus.OK,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps({"delay": _state["delay"]})
        }
    
    elif method == "POST":
        try:
            body = json.loads(event.get("body", "{}"))
            new_delay = int(body.get("delay", _state["delay"]))
            _state["delay"] = clamp_delay(new_delay)
            
            return {
                "statusCode": HTTPStatus.OK,
                "headers": {"Content-Type": "application/json"},
                "body": json.dumps({"delay": _state["delay"]})
            }
        except (ValueError, TypeError, KeyError):
            return {
                "statusCode": HTTPStatus.BAD_REQUEST,
                "body": json.dumps({"error": "Invalid delay value"})
            }
    
    else:
        return {
            "statusCode": HTTPStatus.METHOD_NOT_ALLOWED,
            "body": json.dumps({"error": "Method not allowed"})
        }

File: api/delay.js
// /api/delay.js
let currentDelay = 500;

module.exports = async (req, res) => {
  if (req.method === 'GET') {
    res.json({ delay: currentDelay });
  } else if (req.method === 'POST') {
    try {
      const data = JSON.parse(req.body);
      const { delay } = data;
      
      if (typeof delay === 'number' && delay >= 50 && delay <= 2000) {
        currentDelay = delay;
        res.json({ delay: currentDelay });
      } else {
        res.status(400).json({ error: 'Invalid delay' });
      }
    } catch (e) {
      res.status(400).json({ error: 'Invalid JSON' });
    }
  } else {
    res.status(405).end();
  }
};